frontend/src/components/Canvas/Canvas.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { MutableRefObject, useContext, useEffect, useRef, useState } from "react";
+
+import { Context } from "../../context/appContext";
+import { Line, Position } from "../../model/line";
+import { apiProvider } from "../../services/api";
+import "./canvas.scss";
+
+export function Canvas() {
+    const [
+        {
+            sessionStorageContext: { username, color, roomName },
+            pointerSize,
+            canvasRef,
+            contextRef,
+            socket
+        },
+        dispatch
+    ] = useContext(Context);
+    const [isDrawing, setIsDrawing] = useState<boolean>(false);
+    const divRef = useRef() as MutableRefObject<HTMLDivElement>;
+    const [prevPosition, setPrevPosition] = useState<Position>({ x: 0, y: 0 });
+
+    const sendNewLine = (line: Line) => {
+        socket.emit("newLine", line);
+    };
+
+    const drawLine = (line: Line) => {
+        const { startPosition, endPosition, pointerSize, color } = line;
+        contextRef.current.strokeStyle = color;
+        contextRef.current.lineWidth = pointerSize;
+        contextRef.current.beginPath();
+        contextRef.current.moveTo(startPosition.x, startPosition.y);
+        contextRef.current.lineTo(endPosition.x, endPosition.y);
+        contextRef.current.stroke();
+    };
+
+    useEffect(() => {
+        const canvas = canvasRef.current;
+        canvas.width = divRef.current.offsetWidth - 2;
+        canvas.height = divRef.current.offsetHeight;
+
+        canvas.style.width = `${divRef.current.offsetWidth - 2}px`;
+        canvas.style.height = `${divRef.current.offsetHeight}px`;
+    }, []);
+
+    useEffect(() => {
+        const context = canvasRef.current.getContext("2d");
+
+        apiProvider.getLines(roomName).then((lines) => {
+            lines.forEach((line: Line) => {
+                drawLine(line);
+            });
+        });
+
+        socket.on("drawNewLine", (line: Line) => {
+            drawLine(line);
+        });
+
+        socket.on("repaint", (lines: Line[]) => {
+            dispatch({ type: "CLEAR_CANVAS" });
```

Comment:
You know that this dispatch is not necessairly synchronous before you draw the lines below?